nearpy/tests/engine_tests.py

```diff
@@ -73,5 +73,58 @@ def test_retrieval_sparse(self):
             self.assertEqual(y_data, x_data)
             self.assertAlmostEqual(y_distance, 0.0, delta=delta)
 
+    def test_delete_vector_single_hash(self):
+        dim = 5
+        name_hash = 'testHash'
+        unibucket = UniBucket(name_hash)
+        engine = Engine(dim, lshashes=[unibucket])
+        # Index 20 random vectors (set their data to a unique string)
+        for index in range(20):
+            v = numpy.random.randn(dim)
+            engine.store_vector(v, index)
+        # Create random query vector
+        keys = set()
+        bucket = engine.storage.buckets[name_hash][name_hash]
+        keys = {i[1] for i in bucket}
+
+        engine.delete_vector(15) #delete the vector with key = 15
+
+        new_keys = set()
+        bucket = engine.storage.buckets[name_hash][name_hash]
```

Comment:
You use the same code twice (and two more times in the next test). You should probably think about moving it into method, like `get_keys(engine, hash_name)`